### PR TITLE
lazy load comments earlier

### PR DESCRIPTION
**Pull request by andygnewman**
@GlynnPhillips 

Trigger loading of comments 600px above the comments, rather than default of 250px.

(On a desktop that would be about 1/2 way through the first 'More-on').

### CODE REVIEW

**Comment by GlynnPhillips** [comment]
👍

**Comment by andygnewman** [comment]
Have also added in immediate loading of comments when the request has comment permalink parameters. So will scroll down to that comment.

File: client/main.js [hunk 1]
```diff
@@ -58,13 +58,16 @@ bootstrap(({flags, mainCss}) => {
 			const commentsEl = document.getElementById('comments');
 			const commentsJsLocation = commentsEl.getAttribute('data-comments-js');
 			const commentsCssLocation = commentsEl.getAttribute('data-comments-css');
+			const commentsLazyLoad = commentsEl.getAttribute('data-comments-lazyload') === 'true' ? true : false;
```
Comment (bjfletcher) [inline client/main.js]: simplify this to:

const commentsLazyLoad = commentsEl.getAttribute('data-comments-lazyload') === 'true';

File: server/controllers/article.js [hunk 2]
```diff
@@ -44,6 +44,8 @@ function isPremiumArticle (webUrl) {
 module.exports = function articleV3Controller(req, res, next, content) {
 	let asyncWorkToDo = [];
 
+	content.lazyLoadComments = (req.query['lf-content'] && req.query.hubRefSrc) ? false : true;
```
Comment (bjfletcher) [inline server/controllers/article.js]: equivalent to:

`content.lazyLoadComments = !req.query['lf-content'] || !req.query.hubRefSrc;`

File: views/partials/bottom/comments.html [hunk 3]
```diff
@@ -3,6 +3,7 @@
 		<div
 			id="comments"
 			data-trackable="comments"
+			data-comments-lazyload="{{lazyLoadComments}}"
```
Comment (bjfletcher) [inline views/partials/bottom/comments.html]: `data-comments-lazy-load` (with hyphen in between lazy and load) to be consistent with `lazyLoad` elsewhere.

File: server/controllers/article.js (hunk 2 above)
Comment (andygnewman) [inline server/controllers/article.js]: Agree more simple, but I find the original way I had it to be a little more readable and understandable of the intent.

File: client/main.js (hunk 1 above)
Comment (andygnewman) [inline client/main.js]: Good point.

**Comment by andygnewman** [comment]
Livefyre controls the scroll down - when comments are lazy loaded, the scroll down (and pinpointing of the precise comment) will only occur when the comments have been loaded (ie. when in view, which the user doesn't know).
This PR just overrides the lazy load, so comments are loaded immediately (client side) and then the scroll down occurs - there is still a short time lag, but it at least does what the user expects.

File: server/controllers/article.js (hunk 2 above)
Comment (matthew-andrews) [inline server/controllers/article.js]: great idea^^

**Pull request merged**